src/main/java/com/deloitte/elrr/aggregator/rules/ProcessCompetency.java

```diff
@@ -24,224 +28,268 @@
-			log.info("Competency " + activity.getId() + " created.");
+        Competency competency = null;
+        String activityName = "";
+        String activityDescription = "";
 
-		} catch (AggregatorException | ClassCastException | NullPointerException e) {
-			log.error(e.getMessage());
-			e.printStackTrace();
-			throw e;
-		}
+        try {
 
-		return competency;
-	}
+            activityName = langMapUtil.getLangMapValue(activity.getDefinition().getName());
+            activityDescription = langMapUtil.getLangMapValue(activity.getDefinition().getDescription());
 
-	/**
-	 * @param activity
-	 * @return competency
-	 */
-	private Competency updateCompetency(Competency competency, final Activity activity) {
+            competency = new Competency();
+            competency.setIdentifier(activity.getId());
+            competency.setRecordStatus(StatusConstants.COMPLETED);
+            competency.setFrameworkTitle(activityName);
+            competency.setFrameworkDescription(activityDescription);
+            competencyService.save(competency);
+            String[] strings = { "Competency", activity.getId(), "created." };
+            log.info(String.join(" ", strings));
 
-		log.info("Updating competency.");
+        } catch (AggregatorException | ClassCastException | NullPointerException e) {
+            log.error(e.getMessage());
+            e.printStackTrace();
+            throw e;
+        }
 
-		String activityName = "";
-		String activityDescription = "";
+        return competency;
+    }
 
-		try {
+    /**
+     * @param activity
+     * @return competency
+     */
+    private Competency updateCompetency(Competency competency, final Activity activity) {
 
-			activityName = langMapUtil.getLangMapValue(activity.getDefinition().getName());
-			activityDescription = langMapUtil.getLangMapValue(activity.getDefinition().getDescription());
+        log.info("Updating competency.");
 
-			competency.setRecordStatus(COMPLETED);
-			competency.setFrameworkTitle(activityName);
-			competency.setFrameworkDescription(activityDescription);
-			competencyService.update(competency);
-			log.info("Competency " + activity.getId() + " updated.");
+        String activityName = "";
+        String activityDescription = "";
 
-		} catch (AggregatorException | ClassCastException | NullPointerException | RuntimeServiceException e) {
-			log.error(e.getMessage());
-			e.printStackTrace();
-			throw e;
-		}
+        try {
 
-		return competency;
-	}
+            activityName = langMapUtil.getLangMapValue(activity.getDefinition().getName());
+            activityDescription = langMapUtil.getLangMapValue(activity.getDefinition().getDescription());
 
-	/**
-	 * @param Activity
-	 * @param Person
-	 * @param Competency
-	 * @return PersonalCompetency
-	 */
-	private PersonalCompetency processPersonalCompetency(final Activity activity, Person person,
-			final Competency competency) {
+            competency.setRecordStatus(StatusConstants.COMPLETED);
```

Comment:
If you wish this to be predefined constanty it should be defined in _entities_. Making a custom class here in aggregator and just pushing it over as a string is not ideal. Just define that class in entities and change the type in the entity itself. Possibly use an enum like we do in LearningRecord for the same thing.